community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java

```diff
@@ -229,29 +227,43 @@ protected void loadStorage()
         }
     }
 
+    private void verifyFileSizeAndTruncate() throws IOException
+    {
+        String versionTrailer = buildTypeDescriptorAndVersion( getTypeDescriptor() );
+        int expectedVersionLength = UTF8.encode( versionTrailer ).length;
+        try ( PagedFile pagedFile = pageCache
+                .map( getStorageFileName(), pageCache.pageSize() - pageCache.pageSize() % getRecordSize() ) )
+        {
+            long trailerOffset =
+                    getTrailerOffset( pagedFile, versionTrailer.split( " " )[0] );
+            if ( trailerOffset != -1 && trailerOffset % getRecordSize() == 0 )
+            {
+                writeTrailer( pagedFile, new byte[expectedVersionLength], trailerOffset );
+            }
+            else
+            {
+                setStoreNotOk( new IllegalStateException(
+                        "Misaligned file size " + trailerOffset + " for " + this + ", expected version length:" +
+                        expectedVersionLength ) );
+            }
+        }
+    }
```

Comment:
Not with offset, since that usually is computed externally, in different manners.